templates/workspace_services/guacamole/user_resources/guacamole-azure-linuxvm/terraform/vm_config.sh

```diff
@@ -13,10 +13,12 @@ sudo adduser xrdp ssl-cert
 # Required packages for Docker installation
 sudo apt-get install ca-certificates curl gnupg lsb-release
 # Get Docker Public key from Nexus
-curl -fsSL ${nexus_proxy_url}/repository/docker-public-key/gpg | sudo gpg --dearmor -o /etc/apt/trusted.gpg.d/docker-archive-keyring.gpg
+# shellcheck disable=SC2154
+curl -fsSL "${nexus_proxy_url}"/repository/docker-public-key/gpg | sudo gpg --dearmor -o /etc/apt/trusted.gpg.d/docker-archive-keyring.gpg
 
 # Install desktop environment if image doesn't have one already
-if [ ${install_ui} -eq 1 ]; then
+# shellcheck disable=SC2154
+if [ "${install_ui}" -eq 1 ]; then
```

Comment:
This is very easy to fix without ignoring the warning...
Since install_ui should always get a value you can do something like:
```bash
${install_ui:?}
```
Or if you're unsure just put something that won't change the outcome:
```bash
${install_ui:-0}
```

Reply:
A fair point, because it's in fact token replacement by Terraform (`template_file`) I haven't treated them as standard variables - however the null check you propose above is still probably a good idea in case the script finds its way elsewhere in future. Will do the same for the other examples also

Reply:
Fixed